pkg/virt-controller/watch/snapshot/restore.go

```diff
@@ -598,18 +601,77 @@ func (ctrl *VMRestoreController) getSnapshotContent(vmRestore *snapshotv1.Virtua
 	return vmss, nil
 }
 
-func (ctrl *VMRestoreController) getVM(namespace, name string) (*kubevirtv1.VirtualMachine, error) {
+func (ctrl *VMRestoreController) getVM(namespace, name string) (vm *kubevirtv1.VirtualMachine, exists bool, err error) {
 	objKey := cacheKeyFunc(namespace, name)
 	obj, exists, err := ctrl.VMInformer.GetStore().GetByKey(objKey)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
 	if !exists {
-		return nil, fmt.Errorf("VirtualMachine %s/%s does not exist", namespace, name)
+		return nil, false, nil
+	}
+
+	return obj.(*kubevirtv1.VirtualMachine).DeepCopy(), true, nil
+}
+
+func (ctrl *VMRestoreController) createVM(vmRestore *snapshotv1.VirtualMachineRestore) (*kubevirtv1.VirtualMachine, error) {
+	snapshotContent, err := ctrl.getSnapshotContent(vmRestore)
+	if err != nil {
+		return nil, err
+	}
+
+	vm := snapshotContent.Spec.Source.VirtualMachine.DeepCopy()
+
+	vm, err = patchVM(vm, vmRestore.Spec.Patches)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I'm unsure taking a deep copy of the snapshot VM spec is actually the best thing to do here. If that VM has DataVolumeTemplates, then i think things like DV imports will get kicked off when we might not want them too.

I don't really know what happens when that occurs. I think everything is eventually consistent but it might not be optimal. 

@mhenriks maybe you have thoughts on this?

Reply:
Yeah, great catch @davidvossel!  I don't think this is much of an issue if referenced datavolumes already exist and are populated.  But if they don't, they will be created and that will be bad.

Ideally, the new vm will be created (instead of updated) somewhere around here:  https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/snapshot/restore.go#L517

To make that happen will require the `vmRestoreTarget` struct to support a nil vm.  But I think that's not a big deal.  It's mostly there now to support locking/setting status on the VM while it's being restored and that isn't really necessary when creating a new VM.

I apologize for not catching this issue earlier in my review.

Reply:
Indeed awesome catch!
Thanks @davidvossel for catching that, and thanks @mhenriks for the path forward. I'm also sorry to miss this. I fully agree.